tests/bot/cogs/test_token_remover.py

```diff
@@ -1,131 +1,279 @@
+                    valid_time.assert_called_once_with("y")
+
+    async def test_delete_message(self):
+        """The message should be deleted, and a message should be sent to the same channel."""
+        await TokenRemover.delete_message(self.msg)
+
+        self.msg.delete.assert_called_once_with()
+        self.msg.channel.send.assert_called_once_with(
+            token_remover.DELETION_MESSAGE_TEMPLATE.format(mention=self.msg.author.mention)
         )
 
-        for content, censored_token in cases:
-            with self.subTest(content=content, censored_token=censored_token):
-                self.msg.content = content
-                coroutine = self.cog.on_message(self.msg)
-                with self.assertLogs(logger='bot.cogs.token_remover', level=logging.DEBUG) as cm:
-                    self.assertIsNone(asyncio.run(coroutine))  # no return value
-
-                [line] = cm.output
-                log_message = (
-                    "Censored a seemingly valid token sent by "
-                    "lemon (`42`) in #lemonade-stand, "
-                    f"token was `{censored_token}`"
-                )
-                self.assertIn(log_message, line)
-
-                self.msg.delete.assert_called_once_with()
-                self.msg.channel.send.assert_called_once_with(
-                    DELETION_MESSAGE_TEMPLATE.format(mention='@lemon')
-                )
-                self.bot.get_cog.assert_called_with('ModLog')
-                self.msg.author.avatar_url_as.assert_called_once_with(static_format='png')
-
-                mod_log = self.bot.get_cog.return_value
-                mod_log.ignore.assert_called_once_with(Event.message_delete, self.msg.id)
-                mod_log.send_log_message.assert_called_once_with(
-                    icon_url=Icons.token_removed,
-                    colour=Colour(Colours.soft_red),
-                    title="Token removed!",
-                    text=log_message,
-                    thumbnail='picture-lemon.png',
-                    channel_id=Channels.mod_alerts
-                )
-
-
-class TokenRemoverSetupTests(unittest.TestCase):
-    """Tests setup of the `TokenRemover` cog."""
-
-    def test_setup(self):
-        """Setup of the extension should call add_cog."""
+    @autospec("bot.cogs.token_remover", "LOG_MESSAGE")
+    async def test_format_log_message(self, log_message):
+        """Should correctly format the log message with info from the message and token."""
+        log_message.format.return_value = "Howdy"
+        return_value = TokenRemover.format_log_message(self.msg, "MTIz.DN9R_A.xyz")
+
+        self.assertEqual(return_value, log_message.format.return_value)
+        log_message.format.assert_called_once_with(
+            author=self.msg.author,
+            author_id=self.msg.author.id,
+            channel=self.msg.channel.mention,
+            user_id="MTIz",
+            timestamp="DN9R_A",
+            hmac="xxx",
+        )
```

Comment:
What exactly is the purpose of this specific test being a coroutine method? It's not immediately clear to me.

(FYI: you can have non coroutine test methods defined in an `IsolatedAsyncioTestCase` class)

Reply:
No reason any more. Forgot to remove it.